### ISSUE TITLE
Training Steps

### ISSUE THREAD

**Issue by hnhuang** [AlexeyAB/darknet#536]
Hi Alexey,

In yolov3_voc.cfg, the steps = 40000,45000. However, I still get a lot of weights files in backup directory...

Best

Hongnian

**Comment by AlexeyAB** [comment]
This is normal. Sometimes you want to stop training and continue it later. So you can start from the last weights-file.
`darknet.exe detector train data/voc.data cfg/yolov3-voc.cfg /backup/yolov3-voc_1000.weights`

**Comment by hnhuang** [comment]
What does steps do in yolov3_voc.cfg then? I thought it is a way to control the number of weights files that it generates.

**Comment by AlexeyAB** [comment]
`steps` control when `learning_rate` will be changed according to `policy` and `scales`.

For these params in cfg-file:
```
learning_rate=0.001
burn_in=1000
max_batches = 50200
policy=steps
steps=40000,45000
scales=.1,.1
```

* `burn_in=1000` - before 1000 iterations: `learning_rate` will vary according to this formula : https://github.com/AlexeyAB/darknet/blob/a0dc4d717ab2d95e5e90f5b7b6344e8074b81606/src/network.c#L54

* `steps=40000` - at 40000 iterations:  `learning_rate` will be multiplied by  0.1 (`scales=.1`): https://github.com/AlexeyAB/darknet/blob/a0dc4d717ab2d95e5e90f5b7b6344e8074b81606/src/network.c#L64

* `steps=45000` - at 45000 iterations:  `learning_rate` will be multiplied by  0.1 (`scales=.1`) again: https://github.com/AlexeyAB/darknet/blob/a0dc4d717ab2d95e5e90f5b7b6344e8074b81606/src/network.c#L64

**Comment by hnhuang** [comment]
Thanks. That is very helpful.

**Comment by i-chaochen** [comment]
> `steps` control when `learning_rate` will be changed according to `policy` and `scales`.
> 
> For these params in cfg-file:
> 
> ```
> learning_rate=0.001
> burn_in=1000
> max_batches = 50200
> policy=steps
> steps=40000,45000
> scales=.1,.1
> ```
> 
> * `burn_in=1000` - before 1000 iterations: `learning_rate` will vary according to this formula : https://github.com/AlexeyAB/darknet/blob/a0dc4d717ab2d95e5e90f5b7b6344e8074b81606/src/network.c#L54
> * `steps=40000` - at 40000 iterations:  `learning_rate` will be multiplied by  0.1 (`scales=.1`): https://github.com/AlexeyAB/darknet/blob/a0dc4d717ab2d95e5e90f5b7b6344e8074b81606/src/network.c#L64
> * `steps=45000` - at 45000 iterations:  `learning_rate` will be multiplied by  0.1 (`scales=.1`) again: https://github.com/AlexeyAB/darknet/blob/a0dc4d717ab2d95e5e90f5b7b6344e8074b81606/src/network.c#L64

Is any particular reason why the first 1000 iterations learning rate is so small?
